src/context/AuthContext.tsx

```diff
@@ -216,10 +266,37 @@ export const AuthContextProvider: FC<PropsWithChildren> = ({ children }) => {
             await Promise.all([accessTokenTask, refreshTokenTask]);
         } finally {
             setLoggedIn(false);
+            setClaims(null);
             setUser(null);
         }
     }, [discovery]);
 
+    // Reload method.
+    const reload = useCallback(async () => {
+        // User update part, if invalid tokens are detected here, data is properly reset.
+        try {
+            const accessToken = await SecureStore.getItemAsync("accessToken");
+            if (accessToken) {
+                const [claims, user] = await Promise.all([fetchUserInfo(accessToken), fetchUserSelf(accessToken)]);
```

Comment:
Be aware that the `/Users/:self` endpoint will cache certain data like that pulled from the regsys for the validity of the access token. So in order to force refresh, you would need to refresh your access token first.